### PR TITLE
Add query as property to HttpRequestData

### LINKED ISSUE #502: How  to query HttpRequestData .net 5  (Query) gone!
Hi
This is not an issue at all, but rather a question on how to do something 

In the "old" .net core we could do  ]

    ` var myfield =req.Query["MyField"]`

How do you do it using HttpRequestData now?

Hopefully it will be a quick straight answer 

thanks

### PR DESCRIPTION
**Pull request by madelinegordon**
### Issue describing the changes in this PR

resolves #502 

### Pull request checklist

* [ ] My changes **do not** require documentation changes
  * [ ] Otherwise: Documentation issue linked to PR
* [ ] My changes **should not** be added to the release notes for the next release
  * [x] Otherwise: I've added my notes to `release_notes.md`
* [ ] My changes **do not** need to be backported to a previous version
  * [ ] Otherwise: Backport tracked by issue/PR #issue_or_pr
* [x] I have added all required tests (Unit tests, E2E tests)

### CODE REVIEW

**Comment by madelinegordon** [comment]
Open question - in the case where there are multiple queries with the same key and different values, do we want to expose the values as type `StringValues`, or as a comma-separated string? The protobuf sends that data on the request to the worker as a comma-separated string, and ASP net core exposes multiple values to the same key as a collection of separate strings.

File: src/DotNetWorker.Grpc/Http/GrpcHttpRequestData.cs [hunk 1]
```diff
@@ -100,6 +102,8 @@ public override IReadOnlyCollection<IHttpCookie> Cookies
 
         public override Uri Url => _url ??= new Uri(_httpData.Url);
 
+        public override IDictionary<string, StringValues> Query => _query ??= _httpData.NullableQuery.ToDictionary(q => q.Key, q => new StringValues(q.Value?.Value.ToString().Split(',')));
```
Comment (liliankasem) [inline src/DotNetWorker.Grpc/Http/GrpcHttpRequestData.cs:105]: Do we have a test that tests out this logic?

Comment (madelinegordon) [reply src/DotNetWorker.Grpc/Http/GrpcHttpRequestData.cs:105]: That is a good point - not one that tests multiple values for one key. I will add one.

**Comment by madelinegordon** [comment]
> Open question - in the case where there are multiple queries with the same key and different values, do we want to expose the values as type `StringValues`, or as a comma-separated string? The protobuf sends that data on the request to the worker as a comma-separated string, and ASP net core exposes multiple values to the same key as a collection of separate strings.

synced with @fabiocav - we will use the System.Web.HttpUtility ParseQueryString method rather than parsing ourselves, for parity with how customer are currently parsing the query string off the Uri.

File: src/DotNetWorker.Core/Http/HttpRequestData.cs [hunk 2]
```diff
@@ -62,5 +63,10 @@ public HttpRequestData(FunctionContext functionContext)
         /// </summary>
         /// <returns>The response instance.</returns>
         public abstract HttpResponseData CreateResponse();
+
+        /// <summary>
+        /// Gets the <see cref="Query"/> containing the request query. 
+        /// </summary>
+        public abstract NameValueCollection Query { get; }
```
Comment (fabiocav) [inline src/DotNetWorker.Core/Http/HttpRequestData.cs:70]: Would be good to have this defined as a `virtual` method in this type with the same implementation that was added to `GrpcHttpRequestData`. That would make this simpler and remove the requirement to have derived types implement this property and figure out the proper logic.

File: src/DotNetWorker.Core/Http/HttpRequestData.cs [hunk 3]
```diff
@@ -62,5 +64,16 @@ public HttpRequestData(FunctionContext functionContext)
...
         public abstract HttpResponseData CreateResponse();
+
+        /// <summary>
+        /// Gets the <see cref="Query"/> containing the request query. 
+        /// </summary>
+        public virtual NameValueCollection Query
+        {
+            get
+            {
+                return HttpUtility.ParseQueryString(Url.Query);
```
Comment (kshyju) [inline src/DotNetWorker.Core/Http/HttpRequestData.cs:75]: Every time this prop is accessed, the parse method gets executed. Should we store the results of parse in local variable and return that? (basically moving the logic you have in the GrpcHttpRequestData)?

Comment (madelinegordon) [reply src/DotNetWorker.Core/Http/HttpRequestData.cs:75]: I think that would make sense.

File: src/DotNetWorker.Core/Http/HttpRequestData.cs (hunk 2 above)
Comment (fabiocav) [reply src/DotNetWorker.Core/Http/HttpRequestData.cs]: Discussed the reasons (including breaking/impact) offline. Flagging this as resolved.

**Review by fabiocav — APPROVED** [review]

File: test/DotNetWorkerTests/OutputBindings/OutputBindingsMiddlewareTests.cs [hunk 4]
```diff
@@ -3,6 +3,7 @@
 
 using System;
 using System.Collections.Generic;
+using System.Collections.Specialized;
```
[the remark this answers is not in the document]
Comment (madelinegordon) [reply test/DotNetWorkerTests/OutputBindings/OutputBindingsMiddlewareTests.cs]: removed yesterday :)

**Pull request merged**